install/src/main/resources/aspects.bzl

```diff
@@ -578,28 +566,19 @@ def _print_fields(fields):
     separator = ","
     print(separator.join(fields))
 
-def _get_cpp_target_info(target, ctx):
+def extract_cpp_target_info(target, ctx):
     if CcInfo not in target:
-        return []
+        return None
 
     #TODO: Get copts from semantics
-    copts = _get_target_info(ctx, "copts")
-    defines = _get_target_info(ctx, "defines")
-    linkopts = _get_target_info(ctx, "linkopts")
-
-    linkshared = False
-    if hasattr(ctx.rule.attr, "linkshared"):
-        linkshared = ctx.rule.attr.linkshared
+    copts = getattr(ctx.rule.attr, "copts", [])
```

Comment:
for me u can call `getattr` in the struct constructor, but it's up 2 u

Reply:
good idea -> less mess